File: src/main/java/com/kukininj/PostApp/models/Message.java
package com.kukininj.PostApp.models;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import org.hibernate.validator.internal.util.stereotypes.Lazy;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
public class Message {
    @Id
    @GeneratedValue(generator = "UUID")
    public UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
    @JsonIdentityReference(alwaysAsId = true)
    @JsonProperty("userId")
    public User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
    @JsonIdentityReference(alwaysAsId = true)
    @JsonProperty("transactionId")
    public Transaction transaction;

    public LocalDateTime added;
    public String contents;
}

File: src/main/frontend/src/types/message.ts
import { Message } from "PostAppAPI";
import { SimpleUser } from "./user";

export const SimpleMessage: Message = {
    get id() {
        return Math.random().toString();
    },
    contents: "tresc",
    userId: "00000000-0000-0000-0000-000000000000",
    transactionId: "",
    added: new Date(2020, 2, 2),
}
